### ISSUE TITLE
Argument of #import passed to parser was not a valid GraphQL DocumentNode

### ISSUE THREAD

**Issue by gastonmorixe** [detrohutt/babel-plugin-inline-import-graphql-ast#22]
`document ` is the raw file plain text. which throws 

```Argument of #import "./personFragment.graphql"

query PersonQuery(
  $id: ID!
) {
  person(id: $id) {
    ...personFragment
  }
}
 passed to parser was not a valid GraphQL DocumentNode. You may need to use 'graphql-tag' or another method to convert your operation into a document

parser
    react-apollo.browser.umd.js:229:4
graphql
    react-apollo.browser.umd.js:374:20
<unknown>
    index.js:12:10
loadModuleImplementation
    require.js:212:4
guardedLoadModule
    require.js:146:11
_require
    require.js:130:2
<unknown>
    registerScreens.js:3
loadModuleImplementation
    require.js:212:4
guardedLoadModule
    require.js:146:11
```

```
  function parser(document) {
    var variables, type, name
    invariant$2(
      !!document && !!document.kind,
      'Argument of ' +
        document +
        " passed to parser was not a valid GraphQL DocumentNode. You may need to use 'graphql-tag' or another method to convert your operation into a document"
    )
```

![screen shot 2018-03-09 at 11 55 27 am](https://user-images.githubusercontent.com/637225/37213439-cbb76ac4-2390-11e8-8674-435160325c4a.png)

**Comment by detrohutt** [comment]
What version of my package are you using? If you are using `2.3.x` please try downgrading to `2.2.0` (or `2.1.2` if the error persists). If that doesn't fix the problem you may be experiencing a problem with a "shadowed" variable where you've created another variable or class with the same name as the import and accidentally passed that to the graphql() call instead of the imported variable. You could try `console.log(PersonQuery)` directly above where you're doing `graphql(PersonQuery)` to see what you're actually passing. If you get that far and it's still not fixed, let me know what that console.log returns and we'll go from there.

**Comment by detrohutt** [comment]
Sorry I should've read your description more carefully. Obviously no need to console.log as you already know what you're passing, but still try downgrading as I mentioned so I can see if it is a problem with the new version I just released.

**Comment by gastonmorixe** [comment]
Will try soon. Forgot to say this is inn react native 0.54

**Comment by detrohutt** [comment]
@gastonmorixe Thanks for reporting this. I was able to reproduce it and fix it. Please upgrade to `bable-plugin-inline-import-graphql-ast@^2.3.2`!

**Comment by gastonmorixe** [comment]
Thanks will try later today and report.

On Friday, March 9, 2018, Alexander Roberts <notifications@github.com>
wrote:
> @gastonmorixe Thanks for reporting this. I was able to reproduce it and
fix it. Please upgrade to bable-plugin-inline-import-graphql-ast@^2.3.2!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.<
https://ci3.googleusercontent.com/proxy/0x1ZVmmPH3eUAQeKbdai_8p_qmiZ4ZvXw6b_whR5OJv5D9XY5xf2w1cI8MJdleOQyWy7nV1HKJ06czHJVIWcXOKdkIJEK2d8inkICMnXPXXVG4KgK7Iev4txImZIUF3Y1WOWkW0Ju7Rv7EXhUr32gsh2g4geoA=s0-d-e1-ft#https://github.com/notifications/beacon/AAm5KTFlJzrRkLCHoev0Q-XZ750rY2Qqks5tcsC9gaJpZM4SkaSG.gif
>

-- 
Sent from Gmail Mobile

**Comment by detrohutt** [comment]
@james-ff ping

**Comment by james-ff** [comment]
@detrohutt Thanks for the ping!
Just a note:
`bable-plugin-inline-import-graphql-ast@^2.3.2`
should be
`bable-plugin-inline-import-graphql-ast@^2.3.2`
I got a 404 and took me a while to realise why =P

**Comment by james-ff** [comment]
I've installed 2.3.2 successfully, but I'm still getting the error. I have the package installed as an npm dev dependency, the .babelrc file lists the plugin too. It seems to be having an issue with the fragment import, any ideas? Thanks!

**Comment by james-ff** [comment]
Just applied one of the caveats: "Applying changes to GraphQL files" and it seems to have fixed it (might have been reloading the project etc that fixed it, probably one of the two).

tldr: It seems fixed for me now =]

**Comment by detrohutt** [comment]
Thanks, fixed the typo. And thanks for the update! I was worried my "fix" had broken something. :laughing:

**Comment by gastonmorixe** [comment]
Seems to be solved! I am having cache issues, only after I rename
everything (file names and imports) I see changes. Should I open a new
issue?

On Friday, March 9, 2018, Alexander Roberts <notifications@github.com>
wrote:
> Thanks, fixed the typo. And thanks for the update! I was worried my "fix"
had broken something. 😆
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.<
https://ci5.googleusercontent.com/proxy/NFswM7AN3M7DDBDetk81iWFVsxGkWiWPEtrZXGRZFSrM2JPS3MaY8PPu6VLSU5ZnWa4IlTAEOmh9G5y8TIoCuU-CqfbuYad8u_13W5n8X20UYfsGPulz7LBDk8BNGGojaNPdIme00DsnHgLNNPWBegxyDMHBAw=s0-d-e1-ft#https://github.com/notifications/beacon/AAm5KR2687GmFyDaPOwhZlsF3-d637Ddks5tcs_GgaJpZM4SkaSG.gif
>

-- 
Sent from Gmail Mobile

**Comment by detrohutt** [comment]
See the Caveats section of the README. If you’re still having issues after that you can open another issue.
